libopenage/unit/action.cpp

```diff
@@ -803,6 +803,31 @@ void BuildAction::update_in_range(unsigned int time, Unit *target_unit) {
 	this->frame += time * this->frame_rate / 2.5f;
 }
 
+void BuildAction::on_completion() {
+	if (this->get_target().get()->get_attribute<attr_type::building>().completed < 1.0f) {
+		// The BuildAction was just aborted and we shouldn't look for new buildings
+		return;
+	}
+	this->entity->log(MSG(dbg) << "Done building, searching for new building");
+	auto valid = [this](const TerrainObject &obj) {
+		if (!obj.unit.has_attribute(attr_type::building) ||
+		    obj.unit.get_attribute<attr_type::building>().completed >= 1.0f) {
+			return false;
+		}
+		this->entity->log(MSG(dbg) << "Found unit " << obj.unit.logsource_name());
+		return true;
+	};
+
+	TerrainObject *new_target = find_in_radius(*this->entity->location, valid, 9.0f);
```

Comment:
would be cool if you moved out the `9.0f` as a constexpr `float search_tile_distance` or something so we find it easier when replacing it someday.